src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -93,8 +95,23 @@ class LicenceService(
     }
   }
 
+  fun findLicencesByStaffIdAndStatuses(staffId: Long, statuses: List<LicenceStatus>?): List<LicenceSummary> {
+    val licences: List<LicenceEntity> = if (statuses != null) {
+      licenceRepository.findAllByComStaffIdAndStatusCodeIn(staffId, statuses)
+    } else {
+      licenceRepository.findAllByComStaffId(staffId)
+    }
+    return transformToListOfSummaries(licences)
+  }
+
   private fun offenderHasLicenceInFlight(nomsId: String): Boolean {
-    val inFlight = licenceRepository.findAllByNomsIdAndStatusCodeIn(nomsId, listOf(IN_PROGRESS, SUBMITTED, REJECTED))
+    val inFlight = licenceRepository.findAllByNomsIdAndStatusCodeIn(
+      nomsId,
+      listOf(
+        IN_PROGRESS, SUBMITTED,
+        LicenceStatus.APPROVED, REJECTED
+      )
```

Comment:
Odd needing to specify the LicenceStatus.APPROVED in full? Could include this enum value like the others?